architecture-model-generator/modules/architecture-model-generator-core/src/main/java/io/ballerina/artifactsgenerator/Artifact.java

```diff
@@ -0,0 +1,194 @@
+/*
+ *  Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com)
+ *
+ *  WSO2 LLC. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package io.ballerina.artifactsgenerator;
+
+import io.ballerina.compiler.api.ModuleID;
+import io.ballerina.compiler.api.symbols.ModuleSymbol;
+import io.ballerina.compiler.api.symbols.Symbol;
+import io.ballerina.compiler.syntax.tree.Node;
+import io.ballerina.designmodelgenerator.core.CommonUtils;
+import io.ballerina.runtime.api.utils.IdentifierUtils;
+import io.ballerina.tools.text.LineRange;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * Represents an artifact in the project tree.
+ *
+ * @param id       unique identifier for the artifact
+ * @param location location information of the artifact
+ * @param type     type of the artifact
+ * @param name     symbol name of the artifact
+ * @param accessor accessor of the artifact
+ * @param scope    scope of the artifact (global/local)
+ * @param icon     icon representing the artifact
+ * @param children map of child artifacts (id -> child)
+ * @param module   module name of the artifact
+ * @since 2.3.0
+ */
+public record Artifact(String id, LineRange location, String type, String name, String accessor,
+                       String scope, String icon, String module, Map<String, Artifact> children) {
+
+    private static final Map<String, String> typeCategoryMap = Map.ofEntries(
+            Map.entry(Type.SERVICE.name(), "Entry Points"),
+            Map.entry(Type.AUTOMATION.name(), "Entry Points"),
+            Map.entry(Type.RESOURCE.name(), "Resources"),
```

Comment:
why do we have these two categories `Resources` and `Remote`. Shouldn't this be listed inside the service artifact?

Reply:
The category name is not considered here. These are added as children of the service endpoint.